simulate/sam.go

```diff
@@ -169,3 +176,42 @@ func sequencingError(currSam sam.Sam, alias numbers.BinomialAlias) sam.Sam {
 	}
 	return currSam
 }
+
+func ancientDamage(currSam sam.Sam, ancientAlias numbers.BinomialAlias, geometricParam float64, forward bool, deaminationDistributionSlice []int) sam.Sam {
+	var currRandPos int
+	var distanceToEnd int
+	var foundInMap bool
+	numAncientErrorAttempts := numbers.RandBinomial(ancientAlias)
+	damagedPositions := make(map[int]int, numAncientErrorAttempts)
+	var currError int = 0
+	for currError < numAncientErrorAttempts {
+		distanceToEnd = numbers.RandGeometric(geometricParam)
+		for distanceToEnd >= len(currSam.Seq) {
+			distanceToEnd = numbers.RandGeometric(geometricParam)
+		}
+		if !forward {
+			currRandPos = len(currSam.Seq) - distanceToEnd - 1 // distance from end, instead of distance from start
```

Comment:
At code review we talked about how "ancient damage" is on the DNA molecule and should be detected on both forward and reverse sam reads, unlike "sequencing error".  Does this mean there needs to be code here that says that "if !forward" not only the currRandPos needs to be mutated, but also the matching position on the forward read (and vice versa for "else", mutate matching position on both forward read and reverse read?)